File: problems/doing/1045.py
# 1045 - Triangle Types
def main():
    from sys import stdin, stdout
    lengths = [float(x) for x in stdin.readline().split()]
    lengths.sort()

    A = lengths[2]
    B = lengths[1]
    C = lengths[0]

    not_triangle = A >= B + C
    temp = B*B + C*C

    if not_triangle:
        stdout.write('NAO FORMA TRIANGULO\n')
    elif (A*A == temp):
        stdout.write('TRIANGULO RETANGULO\n')
    elif (A*A > temp):
        stdout.write('TRIANGULO OBTUSANGULO\n')
    elif (A*A < temp):
        stdout.write('TRIANGULO ACUTANGULO\n')
    if (A == B == C):
        stdout.write('TRIANGULO EQUILATERO\n')
    if ((A == B != C) or (B == C != A) or (C == A != B)) and not not_triangle:
        stdout.write('TRIANGULO ISOSCELES\n')

if __name__ == '__main__':
    main()

File: done/1045.c
#include <stdio.h>

int main(void)
{
    float a, b, c;
    scanf("%f %f %f", &a, &b, &c);

    if (b >= a && b >= c)
    {
        a = a + b, b = a - b, a = a - b;
    }
    else if (c >= a && c >= b)
    {
        a = a + c, c = a - c, a = a - c;
    }
    if (c > b)
    {
        b = b + c, c = b - c, b = b - c;
    }

    if (a >= b + c)
    {
        printf("NAO FORMA TRIANGULO\n");
    }
    else
    {
        if (a * a == b * b + c * c)
        {
            printf("TRIANGULO RETANGULO\n");
        }
        if (a * a > b * b + c * c)
        {
            printf("TRIANGULO OBTUSANGULO\n");
        }
        if (a * a < b * b + c * c)
        {
            printf("TRIANGULO ACUTANGULO\n");
        }
        if (a == b && b == c)
        {
            printf("TRIANGULO EQUILATERO\n");
        }
        else if (a == b || b == c || a == c)
        {
            printf("TRIANGULO ISOSCELES\n");
        }
    }

    return 0;
}
